src/app/features/charts/components/median-barchart/median-barchart.component.spec.ts

```diff
@@ -0,0 +1,222 @@
+    component.xAxisLabel = xAxisLabel;
+    component.yAxisLabel = yAxisLabel;
+
+    fixture.detectChanges();
+    element = fixture.nativeElement;
+    const chart = element.querySelector('svg g');
+    return { chart };
+  };
+
+  it('should create', () => {
+    setUp();
+    expect(component).toBeTruthy();
+  });
+
+  it('should not render the chart if there is no data', () => {
+    const { chart } = setUp([]);
+
+    expect(component.data?.length).toEqual(0);
+    expect(element.querySelector('.chart-no-data')).toBeTruthy();
+
+    expect(chart).toBeFalsy();
+  });
+
+  it('should not render the chart if all values are negative', () => {
+    const { chart } = setUp(
+      SMALL_MOCK_DATA.map((obj) => {
+        return { ...obj, medianlogcpm: -1 * obj.medianlogcpm };
+      })
+    );
+
+    expect(element.querySelector('.chart-no-data')).toBeTruthy();
+
+    expect(chart).toBeFalsy();
+  });
+
+  it('should render the chart if there is positive data', () => {
+    const createChartSpy = spyOn(component, 'createChart').and.callThrough();
+    const { chart } = setUp();
+
+    expect(component.data?.length).toEqual(FULL_MOCK_DATA.length);
+    expect(createChartSpy).toHaveBeenCalled();
+    expect(chart).toBeTruthy();
+
+    expect(element.querySelector('.chart-no-data')).toBeFalsy();
+  });
+
+  it('should render the chart axes', () => {
+    const { chart } = setUp();
+
+    expect(chart?.querySelector('.x-axis')).toBeTruthy();
+    expect(chart?.querySelector('.y-axis')).toBeTruthy();
+  });
+
+  it('should render the correct number of bars', () => {
+    const { chart } = setUp();
+
+    expect(chart?.querySelectorAll('rect').length).toEqual(
+      FULL_MOCK_DATA.length
+    );
+  });
+
+  it('should not render bars for negative values', () => {
+    const { chart } = setUp([
+      { ...SMALL_MOCK_DATA[0], medianlogcpm: -0.01 },
+      ...SMALL_MOCK_DATA.slice(1),
+    ]);
+
+    expect(chart?.querySelectorAll('rect').length).toEqual(
+      SMALL_MOCK_DATA.length - 1
+    );
+  });
+
+  it('should render the bar labels', () => {
+    const { chart } = setUp();
+
+    expect(chart?.querySelectorAll('text.bar-labels').length).toEqual(
+      FULL_MOCK_DATA.length
+    );
+  });
+
+  it('should render the labels for both axes', () => {
+    const { chart } = setUp();
+
+    expect(chart?.querySelector('.x-axis-label')?.textContent).toEqual(
+      XAXIS_LABEL
+    );
+    expect(chart?.querySelector('.y-axis-label')?.textContent).toEqual(
+      YAXIS_LABEL
+    );
+  });
+
+  it('should render the meaningful expression threshold', () => {
+    const { chart } = setUp();
+
+    expect(
+      chart?.querySelector('.meaningful-expression-threshold-line')
+    ).toBeTruthy();
+  });
+
+  it('should render the meaningful expression threshold even when all values are small', () => {
+    const { chart } = setUp(SMALL_MOCK_DATA);
+
+    expect(
+      chart?.querySelector('.meaningful-expression-threshold-line')
+    ).toBeTruthy();
+  });
```

Comment:
Currently, the meaningful expression threshold is only shown when the max medianlogcpm value is >= the threshold. In the refactored chart, the threshold is always shown, so that the user has a reference for whether the value is meaningful, even when the values are very small. Happy to change this to match the current behavior, if that's preferred though!

develop | feature
:---:|:---:
<img width="1775" alt="main_ENSG00000202337_lowValue" src="https://github.com/Sage-Bionetworks/Agora/assets/26949006/fd026891-c2e7-4daf-aa60-a751047f5fa3"> | <img width="1772" alt="feature_ENSG00000202337_lowValue" src="https://github.com/Sage-Bionetworks/Agora/assets/26949006/97b6c19c-4699-4efc-9110-cd999c4d4944">

Reply:
This would be a good question for @JessterB

Reply:
This sounds like an improvement to me!

Reply:
I don't see this when running the branch locally, but that's fine.